File: Example/NotificationContent/NotificationViewController.m
#import "NotificationViewController.h"

@implementation NotificationViewController

- (void)viewDidLoad {
    self.contentType = CTNotificationContentTypeContentSlider;  // CTNotificationContentTypeContentSlider is the default, just here for illustration
    [super viewDidLoad];
}

// optional: implement to get user event type data
- (void)userDidPerformAction:(NSString *)action withProperties:(NSDictionary *)properties {
    NSLog(@"user did perform action: %@ with props: %@", action , properties);
}

// optional: implement to get notification response
- (void)userDidReceiveNotificationResponse:(UNNotificationResponse *)response {
    NSLog(@"user did receive notification response: %@:", response);
}

@end

File: ExampleSwiftPM/NotificationContent/NotificationViewController.swift
import CTNotificationContent

class NotificationViewController: CTNotificationViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    // optional: implement to get user event data
    override func userDidPerformAction(_ action: String, withProperties properties: [AnyHashable : Any]!) {
        print("userDidPerformAction \(action) with props \(String(describing: properties))")
    }
    
}
